inference-engine/tests/functional/plugin/cpu/single_layer_tests/nonzero.cpp

```diff
@@ -0,0 +1,166 @@
+        result << "TS=(";
+        for (const auto& shape : inputShape.second) {
+            result << CommonTestUtils::vec2str(shape) << "_";
+        }
+        result << ")_";
+        result << "netPRC=" << netType;
+        result << CPUTestsBase::getTestCaseName(cpuParams);
+        return result.str();
+    }
+protected:
+    void SetUp() override {
+        targetDevice = CommonTestUtils::DEVICE_CPU;
+        NonZeroLayerTestParams basicParamsSet;
+        CPUSpecificParams cpuParams;
+        std::tie(basicParamsSet, cpuParams) = this->GetParam();
+        std::tie(inFmts, outFmts, priority, selectedType) = cpuParams;
+        ElementType netType = ElementType::undefined;
+        InputShape inputShape;
+        std::tie(inputShape, netType) = basicParamsSet;
+
+        init_input_shapes({inputShape});
+        auto inputParams = ngraph::builder::makeDynamicParams(netType, inputDynamicShapes);
+
+        auto nonZero = std::make_shared<ngraph::opset3::NonZero>(inputParams[0]);
+        // I8 was used as a special placeholder during calculating of primitive type if input was U8,
+        // real runtime precision is still U8
+        selectedType = makeSelectedTypeStr("ref_", netType == ElementType::u8 ? ElementType::i8 : netType);
+        inputParams[0]->set_friendly_name("input");
+        function = makeNgraphFunction(netType, inputParams, nonZero, "NonZero");
+    }
+};
+
+TEST_P(NonZeroLayerCPUTest, CompareWithRefs) {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+    run();
+    CheckPluginRelatedResults(executableNetwork, "NonZero");
+}
+
+namespace {
+
+/* CPU PARAMS */
+std::vector<CPUSpecificParams> filterCPUInfoForDevice() {
+    return std::vector<CPUSpecificParams> {CPUSpecificParams{{}, {nc}, {}, {}}};;
+}
+
+const std::vector<ElementType> netPrecisions = {
+        ElementType::f32,
+        ElementType::bf16,
+        ElementType::i32,
+        ElementType::i8,
+        ElementType::u8
+};
+
+std::vector<InputShape> inShapesDynamic = {
+        {
+            //dynamic shape
+            {-1},
+            { //target static shapes
+                {100},
+                {200},
+                {300}
+            }
+        },
+        {
+            //dynamic shape
+            {-1, -1},
+            { //target static shapes
+                {4, 100},
+                {4, 200},
+                {4, 300}
+            }
+        },
+        {
+            //dynamic shape
+            {-1, -1, -1},
+            { //target static shapes
+                {4, 4, 100},
+                {4, 4, 200},
+                {4, 4, 300}
+            }
+        },
+        {
+            //dynamic shape
+            {-1, -1, -1, -1},
+            { //target static shapes
+                {4, 4, 4, 100},
+                {4, 4, 4, 200},
+                {4, 4, 4, 300}
+            }
+        },
+        {
+            //dynamic shape
+            {-1, {1, 10}, -1, {1, 500}},
```

Comment:
could you add test case with fully defined upper bounds please?

Reply:
done